### DISCUSSION TITLE
Is the force-cache header only auto-applied in server components?

### DISCUSSION THREAD

**Discussion by codinginflow** [vercel/next.js#49976] [App Router]
Does NextJS only automatically add the `force-cache` header to any `fetch` request in server components, or also client-side when I fetch data in `useEffect`?

**Comment by icyJoseph** [comment]
I think you might find this section of the docs interesting:

https://nextjs.org/docs/app/building-your-application/routing/linking-and-navigating#client-side-caching-of-rendered-server-components

Specially: 

> Good to know: This client-side cache is different from the server-side [Next.js HTTP cache](https://nextjs.org/docs/app/building-your-application/data-fetching#caching-data).
> 
> The new router has an in-memory client-side cache that stores the rendered result of Server Components (payload). The cache is split by route segments which allows invalidation at any level and ensures consistency across concurrent renders.
> 

I think this article might be a good read too: https://www.inngest.com/blog/5-lessons-learned-from-taking-next-js-app-router-to-production

Finally, let's try it out:

```jsx
"use client";

import { useEffect, useReducer } from "react";

export default function Home() {
  const [, force] = useReducer((x) => !x, false);

  useEffect(() => {
    const controller = new AbortController();
    fetch("https://pokeapi.co/api/v2/pokemon/ditto", {
      signal: controller.signal,
    })
      .then((res) => res.json())
      .then(console.log);

    return () => controller.abort();
  });
  return (
    <section>
      <button onClick={force}>re-render</button>
    </section>
  );
}

```

If we build and run this app, we see in the network tab, that a new request is made every time, so no caching happened there.

**Reply by codinginflow** [reply]
But what you're describing is caching behavior of the router. Is this related to `fetch`? Also, the new caching behavior seems to make it impossible to revalidate pages by navigating back and forth between them. Now you have to refresh the browser tab to see the latest data (even with `revalidate = 0`).

Also, if you use `revalidate = 0`, deduping stops working, right? So now how do you prevent `generateMetadata` and the Page to fetch the same data twice unnecessarily?

**Reply by icyJoseph** [reply]
My snippet shows that the client side fetch is not cached by Next.js

To revalidate paths, that were generated server side, the revalidate pair of functions (revalidatePath and revalidateTag) should do the trick. I have not trained so much with these though. 

I mean, GSP w/ ISR pages could still show stale data, the only way to always show latest was GSSP or client side fetching, that hasn't really changed, though I use force-dynamic. 

I think theres an open discussion of people frustrated with GSP/ISR showing stale data on SPA transitions, under certain conditions I can't remember.

**Reply by codinginflow** [reply]
I want the behavior of getServerSideProps, which fetches fresh data server-side whenever we open the page (including through Links inside the app). The problem is, dynamic rendering in the app directory doesn't show fresh data after navigating and you have to actually refresh the browser tab.

And thank you or your example code snippet!
